lib/host-sp-messages/src/lib.rs

```diff
@@ -175,6 +186,181 @@ pub enum KeyLookupResult {
     MaxResponseLenTooShort,
 }
 
+/// Results for an inventory data request
+///
+/// These **cannot be reordered**; the host and SP must agree on them.
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize, SerializedSize,
+)]
+pub enum InventoryDataResult {
+    Ok,
+    /// The given index is larger than our device count
+    InvalidIndex,
+    /// Communication with the device failed in a way that suggests its absence
+    DeviceAbsent,
+    /// Communication with the device failed in some other way
+    DeviceFailed,
+    /// Failed to serialize data
+    SerializationError,
+}
+
+impl From<HubpackError> for InventoryDataResult {
+    fn from(_: HubpackError) -> Self {
+        Self::SerializationError
+    }
+}
+
+impl From<drv_i2c_api::ResponseCode> for InventoryDataResult {
+    fn from(e: drv_i2c_api::ResponseCode) -> Self {
+        match e {
+            drv_i2c_api::ResponseCode::NoDevice => {
+                InventoryDataResult::DeviceAbsent
+            }
+            _ => InventoryDataResult::DeviceFailed,
+        }
+    }
+}
+
+/// Data payload for an inventory data request
+///
+/// These **cannot be reordered**; the host and SP must agree on them.  New
```

Comment:
Ditto adding a unit test for the serialized values, with apologies because constructing values for each variant on this one looks pretty annoying.

Reply:
Done in 6cf4d21a8969e51f5c56862b00e4b6af6c13f2e3 (and yes, it was annoying!)